### PR TITLE
Allow for spaces in column names from source system

### PR DESCRIPTION
**Pull request by jamiebartlett-kps**
Had to include a few methods from the Apache DDL library and tweak them a bit.

### CODE REVIEW

File: commercedbsync/src/com/sap/cx/boosters/commercedbsync/repository/platform/MigrationHybrisMSSqlPlatform.java [hunk 1]
```diff
@@ -10,27 +10,32 @@
...
+import java.util.Collection;
+import java.util.Iterator;
 import java.util.Map;
 import java.util.Set;
 
 public class MigrationHybrisMSSqlPlatform extends MSSqlPlatform implements HybrisPlatform {
 
     private static final Logger LOG = LoggerFactory.getLogger(MigrationHybrisMSSqlPlatform.class);
 
+    private static final String[] KNOWN_SYSTEM_TABLES = new String[]{"dtproperties"};
```
Comment (lnowakowski) [inline commercedbsync/src/com/sap/cx/boosters/commercedbsync/repository/platform/MigrationHybrisMSSqlPlatform.java:37]: DB Sync is designed to handle only SAP Commerce type system tables, custom tables should be handled by `migration.data.tables.custom` property. Any particular reason for such customization to be hardcoded here?

File: commercedbsync/src/com/sap/cx/boosters/commercedbsync/repository/platform/MigrationHybrisMSSqlPlatform.java [hunk 2]
```diff
@@ -96,7 +101,139 @@ public HybrisMSSqlModelReader(Platform platform) {
...
+                }
+
+                pks.close();
+                return found;
+            } catch (SQLException var6) {
+                throw new DdlUtilsException(var6);
+            }
+        }
+
+        protected void determineAutoIncrementFromResultSetMetaData(Table table, Column[] columnsToCheck) throws SQLException {
```
Comment (lnowakowski) [inline commercedbsync/src/com/sap/cx/boosters/commercedbsync/repository/platform/MigrationHybrisMSSqlPlatform.java:184]: What is the purpose of this method? SAP Commerce is not utilizing MSSQL autoincrement (`IDENTITY`) functionality.

File: commercedbsync/src/com/sap/cx/boosters/commercedbsync/repository/platform/MigrationHybrisMSSqlPlatform.java (hunk 1 above)
Comment (lnowakowski) [reply commercedbsync/src/com/sap/cx/boosters/commercedbsync/repository/platform/MigrationHybrisMSSqlPlatform.java:37]: If this is related with Database Diagrams used in older versions of MSSQL, there is no intent to handle such in core functionality of DB Sync

**Review by lnowakowski — CHANGES_REQUESTED** [review]

**Comment by jamiebartlett-kps** [comment]
Hi @lnowakowski 

Thanks for your comments. In the PR, the only change is actually only the below snippet:

 ```if (this.getPlatform().isDelimitedIdentifierModeOn()) {
    query.append(this.getPlatformInfo().getDelimiterToken());
}

query.append(table.getName());
if (this.getPlatform().isDelimitedIdentifierModeOn()) {
    query.append(this.getPlatformInfo().getDelimiterToken());
}```

To add the delimiter tokens around the outside of the table name after the FROM in the query. The rest of the changes were necessary as the method which required changing was protected inside the Apache DDLUtils Library.

**Comment by lnowakowski** [comment]
Thanks for response @jamiebartlett-kps , can you elaborate a bit more, how common in your case is that SAP Commerce type system attribute has space char in its name? IMO it's not verry often, nor maybe it's not even following general recommendations or naming conventions for DB columns

**Comment by jamiebartlett-kps** [comment]
Agreed that it's uncommon in the source system, but if you're replicating into a target system which isn't empty and also holds tables from other sources, it's possible. This is our use case and we've resolved the issue using this change.

**Comment by lnowakowski** [comment]
> Agreed that it's uncommon in the source system, but if you're replicating into a target system which isn't empty and also holds tables from other sources, it's possible. This is our use case and we've resolved the issue using this change.

Understood, however if you're not migrating data to such table, I'd recommend to exclude it, if exclusions feature is not handling this, during either schema analysis or migration process, it's a different story that should be handled as a bug

**Pull request closed without merging**